### PR TITLE
PI-1705 log init checkoutapi client

### PR DESCRIPTION
**Pull request by chintan-soni-cko**
## Proposed changes

1 Log when CheckoutAPIClient is initialised
2 Added Unit tests for the log events

## Types of changes

What types of changes does your code introduce to frames-android?
_Put an `x` in the boxes that apply_

* [ ] Bugfix (non-breaking change which fixes an issue)
* [X] New feature (non-breaking change which adds functionality)
* [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code._

* [X] Lint and unit tests pass locally with my changes
* [X] I have added tests that prove my fix is effective or that my feature works
* [ ] I have added necessary documentation (if appropriate)

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: android-sdk/src/main/java/com/checkout/android_sdk/CheckoutAPIClient.java [hunk 1]
```diff
@@ -70,7 +72,29 @@ public CheckoutAPIClient(@NonNull Context context, @NonNull String key, @NonNull
...
+    CheckoutAPIClient(Context context, @NonNull String key, @NonNull Environment environment, FramesLogger framesLogger) {
+        this.mContext = context.getApplicationContext();
+        this.mKey = key;
+        this.mEnvironment = environment;
+        this.mLogger = framesLogger;
+        //Send checkoutApiClientInitialisedEvent on initialization of CheckoutAPIClient
+        mLogger.sendCheckoutApiClientInitialisedEvent(mEnvironment);
+    }
+
+    public CheckoutEventLogger getSdkLogger() {
```
Comment (harry-brown-cko) [inline android-sdk/src/main/java/com/checkout/android_sdk/CheckoutAPIClient.java:90]: can this be private?

File: android-sdk/src/main/java/com/checkout/android_sdk/PaymentForm.java [hunk 2]
```diff
@@ -620,4 +622,15 @@ private void sendBillingFormPresentedEvent() {
             }
         });
     }
+
+    // Passing sdkLogger in the initialization of FramesLogger to independently unit test events
+    public CheckoutEventLogger getSdkLogger() {
```
Comment (harry-brown-cko) [inline android-sdk/src/main/java/com/checkout/android_sdk/PaymentForm.java:627]: can this be private?

File: android-sdk/src/test/java/com/checkout/android_sdk/CheckoutApiClientTest.kt [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    internal fun setUp() {
+        sdkLoggerMock = mock(CheckoutEventLogger::class.java)
+        mockFramesLogger = mock(FramesLogger::class.java)
+    }
+
+    @Test
+    fun `test success logging of checkoutApiClientInitialisedEvent for live environment`() {
+        mockFramesLogger.initialise(mock(Context::class.java), Environment.LIVE, sdkLoggerMock)
+
+        val checkoutAPIClientMock = CheckoutAPIClient(
```
Comment (harry-brown-cko) [inline android-sdk/src/test/java/com/checkout/android_sdk/CheckoutApiClientTest.kt:30]: this is no longer mocked. would `checkoutAPIClient` be a better name?

File: android-sdk/src/test/java/com/checkout/android_sdk/logging/FramesLoggingEventsTest.kt [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+        val environment = Environment.SANDBOX
+
+        assertEquals(getExpectedCheckoutApiClientInitialisedEvent(environment).typeIdentifier,
+            FramesLoggingEventDataProvider.logCheckoutApiClientInitialisedEvent(environment).typeIdentifier)
+
+        assertEquals(getExpectedCheckoutApiClientInitialisedEvent(environment).monitoringLevel,
+            FramesLoggingEventDataProvider.logCheckoutApiClientInitialisedEvent(environment).monitoringLevel)
+
+        assertEquals(getExpectedCheckoutApiClientInitialisedEvent(environment).properties[CheckoutApiClientInitEventAttribute.environment],
+            FramesLoggingEventDataProvider.logCheckoutApiClientInitialisedEvent(environment).properties[CheckoutApiClientInitEventAttribute.environment])
```
Comment (harry-brown-cko) [inline android-sdk/src/test/java/com/checkout/android_sdk/logging/FramesLoggingEventsTest.kt:22]: can we test all attributes?

File: android-sdk/src/main/java/com/checkout/android_sdk/logging/FramesLoggingEventDataProvider.kt [hunk 5]
```diff
@@ -0,0 +1,19 @@
+package com.checkout.android_sdk.logging
+
+import com.checkout.android_sdk.Utils.Environment
+import com.checkout.eventlogger.domain.model.MonitoringLevel
+
+object FramesLoggingEventDataProvider {
+
+    fun logCheckoutApiClientInitialisedEvent(mEnvironment: Environment): FramesLoggingEvent {
+        val eventData = mapOf(
+            CheckoutApiClientInitEventAttribute.environment to mEnvironment
```
Comment (harry-brown-cko) [inline android-sdk/src/main/java/com/checkout/android_sdk/logging/FramesLoggingEventDataProvider.kt:10]: `mEnvironment` would refer to a member variable. Shouldn't this just be `environment`?

File: android-sdk/src/main/java/com/checkout/android_sdk/CheckoutAPIClient.java [hunk 6]
```diff
@@ -168,6 +192,11 @@ public CheckoutAPIClient setGooglePayListener(OnGooglePayTokenGenerated listener
     public String getCorrelationID() {
         return mCorrelationID;
     }
+
+    public Environment getEnvironment() {
+        return mEnvironment;
+    }
```
Comment (harry-brown-cko) [inline android-sdk/src/main/java/com/checkout/android_sdk/CheckoutAPIClient.java:198]: can this be private?

**Review by harry-brown-cko — COMMENTED** [review]
mostly great, just a few comments around code style and scoping

File: android-sdk/src/main/java/com/checkout/android_sdk/CheckoutAPIClient.java (hunk 1 above)
Comment (chintan-soni-cko) [reply android-sdk/src/main/java/com/checkout/android_sdk/CheckoutAPIClient.java:90]: Sure.

File: android-sdk/src/main/java/com/checkout/android_sdk/CheckoutAPIClient.java (hunk 6 above)
Comment (chintan-soni-cko) [reply android-sdk/src/main/java/com/checkout/android_sdk/CheckoutAPIClient.java:198]: mEnvironment is private variable in CheckoutApiClient. To support for testing, I have added public methods to expose this variable.

File: android-sdk/src/main/java/com/checkout/android_sdk/logging/FramesLoggingEventDataProvider.kt (hunk 5 above)
Comment (chintan-soni-cko) [reply android-sdk/src/main/java/com/checkout/android_sdk/logging/FramesLoggingEventDataProvider.kt:10]: Sure.

File: android-sdk/src/main/java/com/checkout/android_sdk/CheckoutAPIClient.java (hunk 6 above)
Comment (harry-brown-cko) [reply android-sdk/src/main/java/com/checkout/android_sdk/CheckoutAPIClient.java:198]: I don't think this method is used in any of the tests. Can we remove it?

**Review by harry-brown-cko — APPROVED** [review]

**Pull request merged**